client/messaging_contentstore/tests/test_contactstore.py

```diff
@@ -0,0 +1,227 @@
+"""
+Tests for messaging_contentstore.contentstore.
+"""
+
+from unittest import TestCase
+
+from requests import HTTPError
+from requests.adapters import HTTPAdapter
+from requests_testadapter import TestSession, Resp, TestAdapter
+
+from verified_fake.fake_contentstore import Request, FakeContentStoreApi
+
+from messaging_contentstore.contentstore import ContentStoreApiClient
+
+
+class FakeContentStoreApiAdapter(HTTPAdapter):
+
+    """
+    Adapter for FakeContentStoreApi
+
+    This inherits directly from HTTPAdapter instead of using TestAdapter
+    because it overrides everything TestAdaptor does.
+    """
+
+    def __init__(self, contentstore_api):
+        self.contentstore_api = contentstore_api
+        super(FakeContentStoreApiAdapter, self).__init__()
+
+    def send(self, request, stream=False, timeout=None,
+             verify=True, cert=None, proxies=None):
+        # print request.path_url
+        req = Request(
+            request.method, request.path_url, request.body, request.headers)
+        resp = self.contentstore_api.handle_request(req)
+        response = Resp(resp.body, resp.code, resp.headers)
+        r = self.build_response(request, response)
+        # print r.status_code
+        return r
+
+
+make_messageset_dict = FakeContentStoreApi.make_messageset_dict
+make_message_dict = FakeContentStoreApi.make_message_dict
+make_schedule_dict = FakeContentStoreApi.make_schedule_dict
+
+
+class TestContentStoreApiClient(TestCase):
+    API_URL = "http://example.com/contentstore"
+    AUTH_TOKEN = "auth_token"
+
+    def setUp(self):
+        self.messageset_data = {}
+        self.schedule_data = {}
+        self.message_data = {}
+        self.binary_content_data = {}
+        self.contentstore_backend = FakeContentStoreApi(
+            "contentstore/", self.AUTH_TOKEN,
+            messageset_data=self.messageset_data,
+            schedule_data=self.schedule_data, message_data=self.message_data,
+            binary_content_data=self.binary_content_data)
+        self.contentstore_backend.messagesets.endpoint_data = {}
+        self.contentstore_backend.messages.endpoint_data = {}
+        self.contentstore_backend.schedules.endpoint_data = {}
+        self.contentstore_backend.binary_contents.endpoint_data = {}
+        self.session = TestSession()
+        adapter = FakeContentStoreApiAdapter(self.contentstore_backend)
+        self.session.mount(self.API_URL, adapter)
+        # self.session = Session()
+
+    def make_client(self, auth_token=AUTH_TOKEN):
+        return ContentStoreApiClient(
+            auth_token, api_url=self.API_URL, session=self.session)
+
+    def make_existing_messageset(self, messageset_data):
+        existing_messageset = make_messageset_dict(messageset_data)
+        self.contentstore_backend.messagesets.endpoint_data[
+            existing_messageset[u"id"]] = existing_messageset
```

Comment:
If you don't override the data above, then all these functions should simplify to `self.messageset_data[key] = value`